flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -4550,6 +4550,95 @@ void OmpStructureChecker::Leave(const parser::DoConstruct &x) {
+                    InteropTypeVal)) ==
+                parser::InteropType::Kind::TargetSync) {
+              isTargetSyncOccured = true;
+              ++targetSyncCount;
+            } else {
+              ++targetCount;
+            }
+            if (targetCount >= 2 || targetSyncCount >= 2) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-type may be speciﬁed "
+                          "at most once."_err_en_US);
+            } 
+          }
+          const auto &InteropVar{parser::Unwrap<parser::OmpObject>(std::get<parser::OmpInitClause::InteropVar>(InitClause.v.t))};
+          const auto *name{parser::Unwrap<parser::Name>(InteropVar)};
+          const auto ObjectName{name->ToString()};
+          if(ObjectNameList.empty()) {
+            ObjectNameList.push_back(ObjectName);
+          } else {
+            if(ObjectNameList.end() != std::find(ObjectNameList.begin(),ObjectNameList.end(),ObjectName)) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-var may be speciﬁed for at most one action-clause "
+                          "of each interop construct."_err_en_US);
+            }
+          }
+        },
+        [&](const parser::OmpClause::Depend &DependClause) {
+          isDependClauseOccured = true;
+        },
+        [&](const parser::OmpClause::Destroy &DestroyClause) {
+          const auto &InteropVar{parser::Unwrap<parser::OmpObject>(DestroyClause.v)};
+          const auto *name{parser::Unwrap<parser::Name>(InteropVar)};
+          const auto ObjectName{name->ToString()};
+          if(ObjectNameList.empty()) {
+            ObjectNameList.push_back(ObjectName);
+          } else {
+            if(ObjectNameList.end() != std::find(ObjectNameList.begin(),ObjectNameList.end(),ObjectName)) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-var may be speciﬁed for at most one action-clause "
+                          "of each interop construct."_err_en_US);
+            }
+          }
+        },
+        [&](const parser::OmpClause::Use &UseClause) {
+          const auto &InteropVar{parser::Unwrap<parser::OmpObject>(UseClause.v)};
+          const auto *name{parser::Unwrap<parser::Name>(InteropVar)};
+          const auto ObjectName{name->ToString()};
+          if(ObjectNameList.empty()) {
+            ObjectNameList.push_back(ObjectName);
+          } else {
+            if(ObjectNameList.end() != std::find(ObjectNameList.begin(),ObjectNameList.end(),ObjectName)) {
+              context_.Say(GetContext().clauseSource,
+                          "Each interop-var may be speciﬁed for at most one action-clause "
+                          "of each interop construct."_err_en_US);
+            }
+          }
+        },
+        [&](const auto &) {},
+    },
+    clause.u);
+  }
+  if (isDependClauseOccured && !isTargetSyncOccured) {
```

Comment:
Let's check `targetSyncCount` is `0`, instead of using a new variable `isTargetSyncOccured`
```suggestion
  if (isDependClauseOccured && !targetSyncCount) {
```

Reply:
Addressed the comment on [#1c57d1e](https://github.com/swatheesh-mcw/llvm-project/pull/1/commits/1c57d1ea2b8e7095a834689ad5b832e6a2a84c37)